src/plots/scatterPlot.ts

```diff
@@ -133,6 +134,31 @@ export class Scatter<X, Y> extends XYPlot<X, Y> {
     return drawSteps;
   }
 
+  protected _propertyProjectors(): AttributeToProjector {
+    const propertyToProjectors = super._propertyProjectors();
+
+    const xProjector = Plot._scaledAccessor(this.x());
+    const yProjector = Plot._scaledAccessor(this.y());
+    propertyToProjectors["x"] = xProjector;
+    propertyToProjectors["y"] = yProjector;
+    propertyToProjectors["transform"] = (datum: any, index: number, dataset: Dataset) => {
+        return "translate(" + xProjector(datum, index, dataset) + "," + yProjector(datum, index, dataset) + ")";
+    };
+    propertyToProjectors["d"] = this._constructSymbolGenerator(true);
+
+    return propertyToProjectors;
+  }
+
+  // When generate is true, callback returns generated symbol, otherwise returns symbol generator
+  protected _constructSymbolGenerator(generate: boolean) {
+    const symbolProjector = Plot._scaledAccessor(this.symbol());
+    const sizeProjector = Plot._scaledAccessor(this.size());
+    return (datum: any, index: number, dataset: Dataset) => {
+      const generator = symbolProjector(datum, index, dataset)(sizeProjector(datum, index, dataset));
+      return generate ? generator(null) : generator;
```

Comment:
this is cleaner but bugs me that it returns two different types depending on the param